ble/services/EddystoneService.h

```diff
@@ -185,6 +185,29 @@ class EddystoneService
         return false;
     }
 
+    /**
+     *  Set Eddystone URL Frame information.
+     *  @param[in] power          TX Power in dB measured at 0 meters from the device.
+     *  @param[in] url            Encoded URL
+     *  @param[in] urlAdvPeriodIn How long to advertise the URL frame (measured in # of adv periods)
+     *  @return false on success, true on failure.
+     */
+    bool setURLFrameEncodedData(int8_t power, const char *encodedUrlIn, uint8_t encodedUriInLength, uint32_t urlAdvPeriodIn) {
+        if (0 == urlAdvPeriodIn) {
+            urlIsSet = false;
+            return false;
+        }
+        defaultUrlPower = power;
+        memcpy(defaultUriData, encodedUrlIn, URI_DATA_MAX);
```

Comment:
would strncpy() be better here? I don't like to use memcpy() because it is unconditional. It might be copying memory which isn't related to the encoded URL. This is not a bug; but just an sanity booster.